code/modules/surgery/amputation.dm

```diff
@@ -6,6 +6,25 @@
 	possible_locs = list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG, BODY_ZONE_HEAD)
 	requires_bodypart_type = 0
 
+/datum/surgery/amputation/mechanic
+	name = "Detach mechanical limb"
+	self_operable = TRUE
+	requires_bodypart_type = BODYPART_ROBOTIC
+	possible_locs = list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	target_mobtypes = list(/mob/living/carbon/human)
+	lying_required = FALSE
+	steps = list(
+		/datum/surgery_step/mechanic_open,
+		/datum/surgery_step/open_hatch,
+		/datum/surgery_step/mechanic_unwrench,
+		/datum/surgery_step/prepare_electronics,
+		/datum/surgery_step/sever_limb/mechanic
+		)
+
+/datum/surgery/amputation/mechanic/can_start(mob/user, mob/living/carbon/target)
+	if(!iscarbon(target) || !ispreternis(target)) // This surgery won't work unless you are working on a preterni
+		return 0
+	return 1
```

Comment:
- You can't be a preternis without being a human and by extension, a carbon(i dont think you can even do surgery on non carbons so what the hell?)
- Use FALSE/TRUE instead of 0/1
- If(boolean) return TRUE else return FALSE is a stupid pattern
- You don't need to be a rocket scientist to know what can_start() return ispreternis means, do you really need to comment that, but w/e, more comments is better than not enough comments I GUESS
```suggestion
/datum/surgery/amputation/mechanic/can_start(mob/user, mob/living/carbon/target)
	return ispreternis(target))
```

Reply:
> (i dont think you can even do surgery on non carbons so what the hell?)

Future proofing.

> Use FALSE/TRUE instead of 0/1

okay

> If(boolean) return TRUE else return FALSE is a stupid pattern

Look at the rest of the file. I copied the syntax of what was already here but you are right.

> You don't need to be a rocket scientist to know what can_start() return ispreternis means

Sometimes I wonder if I need to clarify too but then I remember we're trying to make it readable for everyone (different languages and skill level) so I comment anyway. 
Someone can read ispreternis as a single word and be confused etc.

More retard proofing = better imo that's why we started using autodoc I think.